test/simulator/timestamp/src/session_simulator.cpp

```diff
@@ -36,49 +36,80 @@
 #include "error_simulator.h"
 #include "timestamp_manager.h"
 
-session_simulator::session_simulator()
-    : _has_commit_ts(false), _ts_round_prepared(false), _ts_round_read(false), _txn_running(false),
-      _commit_ts(0), _durable_ts(0), _first_commit_ts(0), _prepare_ts(0), _read_ts(0)
-{
-}
+session_simulator::session_simulator() : _txn_running(false) {}
 
 int
 session_simulator::begin_transaction(const std::string &config)
 {
     /* Make sure that the transaction from this session isn't running. */
     assert(!_txn_running);
 
+    /* Reset timestamps */
+    _commit_ts = 0;
+    _durable_ts = 0;
+    _first_commit_ts = 0;
+    _prepare_ts = 0;
+    _read_ts = 0;
+    _has_commit_ts = false;
+    _ts_round_prepared = false;
+    _ts_round_read = false;
+
     timestamp_manager *ts_manager = &timestamp_manager::get_timestamp_manager();
     std::map<std::string, std::string> config_map;
 
     ts_manager->parse_config(config, config_map);
 
     /* Check if the read or prepared timestamp should be rounded up. */
     auto pos = config_map.find("roundup_timestamps");
-    _ts_round_prepared = false;
-    _ts_round_read = false;
     if (pos != config_map.end()) {
         if (pos->second.find("read=true"))
             _ts_round_read = true;
         if (pos->second.find("prepared=true"))
             _ts_round_prepared = true;
-
         config_map.erase(pos);
     }
 
-    /* Set the read timestamp if it provided. */
+    /* Set and validate the read timestamp if it provided. */
     pos = config_map.find("read_timestamp");
     if (pos != config_map.end()) {
+        WT_SIM_RET(ts_manager->validate_hex_value(pos->second, "read timestamp"));
         uint64_t read_ts = ts_manager->hex_to_decimal(pos->second);
-        if (read_ts == 0)
-            WT_SIM_RET_MSG(EINVAL, "Illegal read timestamp: zero not permitted.");
-        config_map.erase(pos);
         set_read_timestamp(read_ts);
+        config_map.erase(pos);
     }
 
-    _txn_running = true;
+    /*
+     * For now, the simulator only supports roundup_timestamp and read_timestamp in the config
+     * string for begin_transaction(). Hence, we ignore ignore_prepare, isolation, name,
+     * no_timestamp, operation_timeout_ms, priority and sync.
+     */
+    pos = config_map.find("ignore_prepare");
```

Comment:
Can we create a vector of ignored settings and loop through this vector?

Reply:
I prefer keeping it the way it is, looks more readable that way.